### DISCUSSION TITLE
Customizing Error Messages in Validation Rules

### DISCUSSION THREAD

**Discussion by chamyeongdo** [laravel/framework#48904] [General]
I'm trying to create custom error messages.

For instance, there's a rule called 'different'. I was able to customize the error message for 'different' as follows:

> 'different' => ':Attribute and :other must be different.',

Reference gist: https://gist.github.com/johnhout/055d2c039cfb7574d657#file-validation-php-L41

I used the `:other` parameter to write the error message. However, I haven't been able to find any reference in the official documentation or in the source code that explains how the `:other` value is provided.

Also, for the `dimensions` rule, I noticed that parameters like `:max_width`, `:max_height` are provided.

But I couldn't find any information about this in any part of the official documentation either.

**Comment by shaedrich** [comment]
At some point, [`\Illuminate\Validation\Concerns\ReplacesAttributes::replaceDifferent()`](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Validation/Concerns/ReplacesAttributes.php#L93-L105) is called
That, in return, calls [`\Illuminate\Validation\Concerns\ReplacesAttributes::replaceSame()`](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Validation/Concerns/ReplacesAttributes.php#L674-L686), which replaces `:other`

Hope, that helps!
